internal/display/bufwindow.go

```diff
@@ -406,6 +406,9 @@ func (w *BufWindow) displayBuffer() {
 					mb, left, found := b.FindMatchingBrace(bp, curLoc)
 					if found {
 						matchingBraces = append(matchingBraces, mb)
+						if b.Settings["matchbracestyle"].(string) != "highlight" {
+							matchingBraces = append(matchingBraces, curLoc)
+						}
```

Comment:
This doesn't work, since `curLoc` is added to `matchingBraces` below at line 413 regardless of the above check. Also it causes unwanted extra underlining if `left` is true and matchbracestyle is set to underlined (since both curLoc-1 and curLoc are added to `matchingBraces` in this case).

Please take a look again at my original patch (and please test your changes).

Reply:
Oops, that wasn't very smart of me. Didn't notice anything was wrong while testing in `kitty` with custom cursor colors. Should be good now.